pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -111,9 +115,9 @@ type ruleCache struct {
 	// It is a mapping from group name to a set of Pods.
 	addressSetByGroup map[string]v1beta1.GroupMemberPodSet
 
-	policySetLock sync.RWMutex
-	// policySet is a set to store NetworkPolicy UID strings.
-	policySet sets.String
+	policyMapLock sync.RWMutex
+	// policyMap is a map using NetworkPolicy UID as the key.
+	policyMap map[string]*networkPolicyReference
```

Comment:
@tnqn : I changed this set to map for easier lookup of NetworkPolicy from ruleCache. Let me know if you think we should use other approaches.

Reply:
Good suggestion. Changed.